pkg/cmd/connector/update/update.go

```diff
@@ -35,85 +41,184 @@ func NewUpdateCommand(f *factory.Factory) *cobra.Command {
 		Long:    f.Localizer.MustLocalize("connector.update.cmd.longDescription"),
 		Example: f.Localizer.MustLocalize("connector.update.cmd.example"),
-		Hidden:  true,
 		Args:    cobra.NoArgs,
+		Hidden:  true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			validOutputFormats := flagutil.ValidOutputFormats
 			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
 				return flagutil.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
 			}
 
-			return runUpdateCommand(opts)
+			return runUpdate(opts)
 		},
 	}
-
-	flags := connectorcmdutil.NewFlagSet(cmd, f)
-	flags.StringVar(&opts.file, "file", "", f.Localizer.MustLocalize("connector.file.flag.description"))
-	flags.AddConnectorID(&opts.id)
+	flags := flagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.name, "name", "", f.Localizer.MustLocalize("connector.flag.name.description"))
+	flags.StringVar(&opts.namespaceID, "namespace-id", "", f.Localizer.MustLocalize("connector.flag.kafkaID.description"))
+	flags.StringVar(&opts.kafkaID, "kafka-id", "", f.Localizer.MustLocalize("connector.flag.namespaceID.description"))
 	flags.AddOutput(&opts.outputFormat)
 
 	return cmd
-
 }
 
-func runUpdateCommand(opts *options) error {
-	f := opts.f
+func runUpdate(opts *options) error {
+
+	conn, err1 := opts.f.Connection()
+	if err1 != nil {
+		return err1
+	}
+
+	api := conn.API()
+
+	connector, err2 := contextutil.GetCurrentConnectorInstance(&conn, opts.f)
+	if err2 != nil {
+		connector, _ = connectorutil.InteractiveSelect(conn, opts.f)
+	}
+
+	patch, err3 := setValuesInInteractiveMode(connector, opts, &conn)
+	if err3 != nil {
+		return err3
+	}
+
+	connectorSpecification, err4 := askForConnectorDetails(&api, connector, opts)
+	if err4 != nil {
+		return err4
+	}
+
+	patch["connector"] = connectorSpecification
+
+	a := api.ConnectorsMgmt().ConnectorsApi.PatchConnector(opts.f.Context, connector.GetId())
+	a = a.Body(patch)
+	updated, httpRes, err := a.Execute()
+	if httpRes != nil {
+		defer httpRes.Body.Close()
+	}
+
+	if apiErr := connectorerror.GetAPIError(err); apiErr != nil {
+		return opts.f.Localizer.MustLocalizeError("connector.type.update.error.other", localize.NewEntry("Error", apiErr.GetReason()))
+	}
 
-	var conn connection.Connection
-	conn, err := f.Connection()
 	if err != nil {
 		return err
 	}
 
-	var specifiedFile *os.File
-	if opts.file == "" {
-		file, err1 := util.CreateFileFromStdin()
-		if err1 != nil {
-			return err
+	if err = dump.Formatted(opts.f.IOStreams.Out, opts.outputFormat, updated); err != nil {
+		return err
+	}
+
+	opts.f.Logger.Info(opts.f.Localizer.MustLocalize("connector.update.info.success"))
+
+	return nil
+}
+
+func setValuesInInteractiveMode(connector *connectormgmtclient.Connector, opts *options, conn *connection.Connection) (map[string]interface{}, error) {
+
+	patch := make(map[string]interface{})
+
+	if opts.name == "" {
```

Comment:
This feels like dead code. Any *valid* connector cannot have name as empty.
Also logic is strange as if I would like to change name I would not be asked for it etc.